test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SqlProjects/SqlProjectsServiceTests.cs

```diff
@@ -153,6 +153,43 @@ await service.HandleDeleteSqlObjectScriptRequest(new SqlProjectScriptParams()
             Assert.AreEqual(0, service.Projects[projectUri].SqlObjectScripts.Count);
             Assert.IsFalse(File.Exists(scriptFullPath));
         }
+
+        [Test]
+        public async Task TestSqlCmdVariablesAddDelete()
+        {
+            SqlProjectsService service = new();
+            string projectUri = await service.CreateSqlProject();
+
+            Assert.AreEqual(0, service.Projects[projectUri].SqlCmdVariables.Count, "Baseline number of SQLCMD variables not as expected");
+
+            // Validate adding a SQLCMD variable
+            MockRequest<ResultStatus> requestMock = new();
+
+            const string variableName = "TestVarName";
+
+            await service.HandleAddSqlCmdVariableRequest(new AddSqlCmdVariableParams()
+            {
+                ProjectUri = projectUri,
+                Name = variableName,
+                DefaultVault = "TestVarDefaultValue",
+                Value = "TestVarValue"
+            }, requestMock.Object);
+
+            Assert.IsTrue(requestMock.Result.Success, "Successful add request result expected");
+            Assert.AreEqual(1, service.Projects[projectUri].SqlCmdVariables.Count, "Number of SQLCMD variables after addition not as expected");
+            Assert.IsTrue(service.Projects[projectUri].SqlCmdVariables.Contains(variableName), $"List of SQLCMD variables expected to contain {variableName}");
+
+            // Validate deleting a SQLCMD variable
+            requestMock = new();
+            await service.HandleDeleteSqlCmdVariableRequest(new DeleteSqlCmdVariableParams()
```

Comment:
please also add a test for trying to delete a sqlcmd variable that doesn't exist and adding a duplicate sqlcmd variable